app/src/main/java/org/mozilla/fenix/compose/tabstray/TabGridItem.kt

```diff
@@ -0,0 +1,336 @@
+                        backgroundColor = FirefoxTheme.colors.layer2,
+                        isContentRtl = BidiFormatter.getInstance().isRtl(tab.content.title)
+                    ) {
+                        Text(
+                            text = tab.content.title,
+                            fontSize = 14.sp,
+                            maxLines = 1,
+                            softWrap = false,
+                            style = TextStyle(
+                                color = FirefoxTheme.colors.textPrimary,
+                                textDirection = TextDirection.Content
+                            )
+                        )
+                    }
+
+                    Icon(
+                        painter = painterResource(id = R.drawable.mozac_ic_close),
+                        contentDescription = stringResource(id = R.string.close_tab),
+                        tint = FirefoxTheme.colors.iconPrimary,
+                        modifier = Modifier
+                            .clickable { onCloseClick(tab) }
+                            .size(24.dp)
+                            .align(Alignment.CenterVertically)
+
+                    )
+                }
+                Divider(
+                    color = FirefoxTheme.colors.borderPrimary,
+                    thickness = 1.dp
+                )
+
+                Thumbnail(
+                    tab = tab,
+                    multiSelectionSelected = multiSelectionSelected,
+                )
+            }
+        }
+
+        MediaImage(
+            tab = tab,
+            onMediaIconClicked = { onMediaClick(tab) },
+            modifier = Modifier
+                .align(Alignment.TopStart)
+        )
+    }
+}
+
+/**
+ * Displays the [content] with the right edge fading.
+ *
+ * @param modifier [Modifier] for the container.
+ * @param fadeWidth Length of the fading edge.
+ * @param backgroundColor Color of the background shown under the content.
+ * @param isContentRtl Whether or not the content should be displayed Right to Left
+ * @param content The content whose right edge must be faded.
+ */
+@Composable
+private fun FadingEndEdge(
+    modifier: Modifier = Modifier,
+    fadeWidth: Dp = 25.dp,
+    backgroundColor: Color = Color.Transparent,
+    isContentRtl: Boolean = false,
+    content: @Composable () -> Unit
+) {
+    // List of colors defining the direction of the fade effect
+    val colorList = listOf(Color.Transparent, backgroundColor)
+    CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Ltr) {
+        Box(modifier) {
+            content()
+            Spacer(
+                Modifier
+                    .width(fadeWidth)
+                    .fillMaxHeight()
+                    .align(
+                        if (isContentRtl) {
+                            Alignment.CenterStart
+                        } else {
+                            Alignment.CenterEnd
+                        }
```

Comment:
Is this to make sure that the edge blur is always on the right-hand side?

Reply:
This was added to make sure the fade effect is applied on the correct side (`Alignment.CenterEnd` would apply the fade based on the local layout direction). In the current `TextView` [implementation](https://github.com/mozilla-mobile/fenix/blob/bbc835ad02057eb2b09923713f88459e515cb3d0/app/src/main/res/layout/tab_tray_grid_item.xml#L57), the fade effect is applied according to the content's direction.